components/browser/search/src/main/java/mozilla/components/browser/search/suggestions/SearchSuggestionClient.kt

```diff
@@ -0,0 +1,37 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.search.suggestions
+
+import mozilla.components.browser.search.SearchEngine
+
+/**
+ * Async function responsible for taking a URL and returning the results
+ */
+typealias SearchSuggestionFetcher = suspend (url: String) -> String?
+
+/**
+ *  Provides an interface to get search suggestions from a given SearchEngine.
+ */
+class SearchSuggestionClient private constructor(
+    private val searchEngine: SearchEngine,
+    private val fetcher: SearchSuggestionFetcher
+) {
+    companion object {
+        fun create(searchEngine: SearchEngine, fetcher: SearchSuggestionFetcher): SearchSuggestionClient? {
+            if (!searchEngine.canProvideSearchSuggestions) { return null }
+
+            return SearchSuggestionClient(searchEngine, fetcher)
+        }
+    }
+    /**
+     * Gets search suggestions for a given query
+     */
+    suspend fun getSuggestions(query: String): List<String>? {
+        val suggestionsURL = searchEngine.buildSuggestionsURL(query) ?: return null
+        val parser = selectParser(searchEngine)
```

Comment:
I think renaming this to `selectResponseParser` would make this easier to read.

Reply:
@csadilek Great comments. Will revise!